File: core/src/jvmMain/java/com/xebia/functional/xef/conversation/jvm/Description.java
package com.xebia.functional.xef.conversation.jvm;

import kotlinx.serialization.SerialInfo;

import java.lang.annotation.*;

@Target({ElementType.TYPE, ElementType.ANNOTATION_TYPE, ElementType.FIELD, ElementType.METHOD, ElementType.PARAMETER})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@SerialInfo
public @interface Description {
    String value();
}

File: core/src/commonMain/kotlin/com/xebia/functional/xef/conversation/Description.kt
package com.xebia.functional.xef.conversation

import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.SerialInfo

/**
 * Description of a property to enhance the LLM prompt chances to get the data in the right format
 */
@OptIn(ExperimentalSerializationApi::class)
@SerialInfo
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.PROPERTY, AnnotationTarget.FIELD)
annotation class Description(val value: String)
